models/revision.go

```diff
@@ -0,0 +1,210 @@
+// Copyright 2020 The Gitea Authors. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+package models
+
+import (
+	"sort"
+
+	"code.gitea.io/gitea/modules/git"
+	"code.gitea.io/gitea/modules/timeutil"
+)
+
+// Revision Tracking information about PR revisions.
+type Revision struct {
+	ID                int64              `xorm:"pk autoincr"`
+	Commit            string             `xorm:"NOT NULL"`
+	NumberOfCommits   int64              `xorm:"NOT NULL"`
+	PrevChangesetHead string             `xorm:"-"`
+	UserID            int64              `xorm:"NOT NULL"`
+	User              *User              `xorm:"-"`
+	PRID              int64              `xorm:"pr_id NOT NULL"`
+	PR                *PullRequest       `xorm:"-"`
+	Index             int64              `xorm:"NOT NULL"`
+	Created           timeutil.TimeStamp `xorm:"NOT NULL"`
+}
+
+// GetRevision get the revision by pr and revision index.
+func GetRevision(pr *PullRequest, index int64) (*Revision, error) {
+	rev := &Revision{PRID: pr.ID, Index: index}
+	has, e := x.Get(rev)
+
+	if e != nil {
+		return nil, e
+	}
+
+	if has {
+		user, err := GetUserByID(rev.UserID)
+		if err != nil {
+			return nil, err
+		}
+		rev.User = user
+		pr, err := GetPullRequestByID(rev.PRID)
+		if err != nil {
+			return nil, err
+		}
+		rev.PR = pr
+		if index > 1 {
+			prevrev := &Revision{PRID: pr.ID, Index: index - 1}
+			has, e = x.Get(prevrev)
+
+			if e != nil {
+				return nil, e
+			}
+
+			if has {
+				rev.PrevChangesetHead = prevrev.Commit
+			}
+		}
+		return rev, nil
+	}
```

Comment:
Not suggest doing these things in get function, suggest adding other functions to do it. Thanks. 
By the way, if user does not exist, should not return error, use ``NewGhostUser()`` is better.

Reply:
Refactored. The user/ghostuser is fixed.